### ISSUE TITLE
BLD: Enable PEP 517

### ISSUE THREAD

**Issue by fangchenli** [pandas-dev/pandas#39415]
After #38852, we could enable PEP 517 by modifying PATH in setup.py or use non-vendored versioneer. See https://github.com/python-versioneer/python-versioneer/issues/193 for details.

**Comment by andriyor** [comment]
I will recommend switching to setuptools_scm how it was done in matplotlib https://github.com/matplotlib/matplotlib/pull/18971

**Comment by fangchenli** [comment]
> I will recommend switching to setuptools_scm how it was done in matplotlib [matplotlib/matplotlib#18971](https://github.com/matplotlib/matplotlib/pull/18971)

We hope to get the complete git commit hash, but not sure how to do that using `setuptools_scm`. I opened an issue https://github.com/pypa/setuptools_scm/issues/577 for this.

**Comment by jules-ch** [comment]
We can retrieve full commit hash, using setuptools-scm with the following config:

```toml

[tool.setuptools_scm]
git_describe_command = [
            "git",
            "describe",
            "--dirty",
            "--tags",
            "--abbrev=40",
            "--long",
            "--match",
            "*[0-9]*",
        ]
```

This will overide the default git describe command used by setuptools-scm to get the full (40 digits long hex hash).
Then writing a template to generate the same output as today in `_version.py` should work.

**Comment by fangchenli** [comment]
@jules-ch Thanks for this. I posted the link to your comment in the corresponding `setuptools_scm` issue. Hope they can incorporate this to `setuptools_scm ` as a simple setting. 

And in the meantime, a PR is welcome if you want to give this a try^^

**Comment by nathanjmcdougall** [comment]
It is worth noting that as of v0.24 versioneer can be non-vendored, closing this issue:
https://github.com/python-versioneer/python-versioneer/issues/193
If the plan is to move away from versioneer altogether, it may be worth updating this comment accordingly:
https://github.com/pandas-dev/pandas/blob/75429df91ecafff0748816c414aea0a00eeb25b0/pyproject.toml#L10-L12

**Comment by fangchenli** [comment]
closed by #49115
